src/LibChorus/VcsDrivers/Mercurial/HgRepository.cs

```diff
@@ -1475,10 +1471,28 @@ private static void SetExtensions(IniDocument doc, IEnumerable<KeyValuePair<stri
 			}
 		}
 
-		internal static bool CheckExtensions(IniDocument doc, IEnumerable<KeyValuePair<string, string>> extensionDeclarations)
+		/// <summary/>
+		/// <returns>
+		/// Returns true if the extensions in the repo contain only the extensions declared in the given map.
+		///</returns>
+		internal static bool CheckExtensions(IniDocument doc, Dictionary<string, string> extensionDeclarations)
```

Comment:
I believe that the mercurial.ini in the working directory will override the global one. We formerly used a deprecated extension that Mercurial 3 will no longer let us use. That is why it is important that we remove extensions that we don't currently expect. Even if the global settings somehow came into play, they wouldn't actually have settings that are critical to remove or mercurial would be broken system wide.